File: csharp/Request.cs
using NUnit.Framework;
using System;
using System.Collections.Generic;

namespace WikiSearch
{
	public class Request
	{
		internal readonly string Uri;
		internal readonly Dictionary<string, string> Data;
		internal readonly string RequestType;

		public Request(string request_type, string uri, Dictionary<string,string> hashMap)
		{
			this.RequestType = request_type;
			this.Uri = uri;
			this.Data = hashMap;
		}

		public Request(string request_type, string uri) : this(request_type, uri, new Dictionary<string, string>())
		{
		}
	}
}

File: java/src/Request.java
import java.util.HashMap;

class Request {
	private String uri;
	private HashMap<String, String> data;
	private String request_type;

	public Request(String request_type, String uri, HashMap<String,String> hashMap) {
		this.request_type = request_type;
		this.uri = uri;
		this.data = hashMap;
	}

	public Request(String request_type, String uri) {
		this(request_type, uri, new HashMap<String, String>());
	}

	public String request_type() {
		return this.request_type;
	}

	public HashMap<String, String> data() {
		return this.data;
	}

	public String uri() {
		return this.uri;
	}
}
